src/plus/integrations/providers/github.ts

```diff
@@ -317,3 +318,53 @@ export class GitHubEnterpriseIntegration extends GitHubIntegrationBase<SelfHoste
 		return super.connect(source);
 	}
 }
+
+export type GitHubPullRequestURLIdentity = {
+	ownerAndRepo?: string;
+	prNumber?: string;
+};
+
+export function getPullRequestIdentityValuesFromSearch(search: string): GitHubPullRequestURLIdentity {
+	let ownerAndRepo: string | undefined = undefined;
+	let prNumber: string | undefined = undefined;
+
+	let match = search.match(/([^/]+\/[^/]+)\/pull\/(\d+)/); // with org and rep name
+	if (match != null) {
+		ownerAndRepo = match[1];
+		prNumber = match[2];
+	}
+
+	if (prNumber == null) {
+		match = search.match(/(?:\/|^)pull\/(\d+)/); // without repo name
+		if (match != null) {
+			prNumber = match[1];
+		}
+	}
+
+	if (prNumber == null) {
+		match = search.match(/(?:#|\/)(\d+)/); // any number starting with # or /
+		if (match != null) {
+			prNumber = match[1];
+		}
+	}
+
+	return { ownerAndRepo: ownerAndRepo, prNumber: prNumber };
+}
+
+export function doesPullRequestSatisfyGitHubRepositoryURLIdentity(
+	pr: LaunchpadPullRequest | undefined,
```

Comment:
Including Launchpad-specific types here muddles the purity and intent of the provider - instead, since this appears to be a utility fn, we can probably move it to `pullRequest.ts`, and either now or in the near future, make the parameters less GitHub-specific.

Reply:
@axosoft-ramint 

Fixed. And it's possible to use `EnrichablePullRequest` here instead of `LaunchpadPullRequest`.